### PR TITLE
Fix potential dereferencing null pointer.

### PR DESCRIPTION
**Pull request by raniervf**
Malloc can return NULL.

### CODE REVIEW

File: src/driver/interface/maildriver.c [hunk 1]
```diff
@@ -52,7 +52,9 @@ mailsession * mailsession_new(mailsession_driver * sess_driver)
   int r;
   
   session = malloc(sizeof(* session));
-  
+  if (session == NULL)
+    return NULL;
```
Comment (dinhviethoa) [inline src/driver/interface/maildriver.c:56]: could you instead use:
`goto err;`
and add a `err:` label at the bottom before `return NULL`?

Comment (dinhviethoa) [reply src/driver/interface/maildriver.c:56]: That's the pattern that's generally used in the code for error handling & proper clean up.

Comment (raniervf) [reply src/driver/interface/maildriver.c]: Sure. Done.

**Comment by dinhviethoa** [comment]
You still need to add a label at the end.

**Comment by raniervf** [comment]
Done.

**Pull request merged**